adserver/forms.py

```diff
@@ -554,6 +556,170 @@ class Meta:
+        helper.layout = Layout(
+            Fieldset(
+                "",
+                Field("name"),
+                Div(
+                    Div("start_date", css_class="form-group col-lg-6"),
+                    Div("end_date", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                Div(
+                    PrependedText(
+                        "budget",
+                        "$",
+                        min=0,
+                        data_bind="textInput: budget",
+                    ),
+                ),
+                Field("note"),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Advertisements"),
+                Field(
+                    "advertisements",
+                    template="adserver/includes/widgets/advertisement-form-option.html",
+                ),
+                css_class="my-3",
+            ),
+            Submit("submit", _("Request a new flight")),
+            HTML(
+                "<p class='form-text small text-muted'>"
+                + str(
+                    _(
+                        "Your flight will not start automatically. Your account manager will be notified to review your ads and targeting."
+                    )
+                )
+                + "</p>"
+            ),
+        )
+        return helper
+
+    def save(self, commit=True):
+        assert commit, "Delayed saving is not supported on this form"
+
+        # This is already the default but we are setting it explicitly to be defensive
+        self.instance.live = False
+
+        if self.old_flight:
+            # Copy fields not in the form from the old flight if possible
+            # Otherwise, the account manager will need to set these
+            fields = (
+                "targeting_parameters",
+                "priority_multiplier",
+                "cpm",
+                "cpc",
+                "sold_impressions",
+                "sold_clicks",
+            )
+            for field in fields:
+                setattr(self.instance, field, getattr(self.old_flight, field))
+
+        # We must set the campaign
+        self.instance.campaign = (
+            self.old_flight.campaign
+            if self.old_flight
+            else self.advertiser.campaigns.all().first()
+        )
+
+        instance = super().save(commit)
+
+        # Duplicate the advertisements into the new flight
+        if "advertisements" in self.cleaned_data:
+            for ad in self.cleaned_data["advertisements"]:
+                new_ad = ad.__copy__()
+                new_ad.flight = instance
+                new_ad.live = True
+                new_ad.save()  # Automatically gets a new slug
+
+        return instance
+
+    class Meta:
+        model = Flight
+
+        fields = (
+            "name",
+            "start_date",
+            "end_date",
+        )
+        widgets = {
```

Comment:
Interesting that this is required. Does it default to a time & date, and we just want to set it to a date?

Reply:
Without this, you will not get a date widget at all. The `type=date` is what does that. We aren't using a JS date widget but rather an HTML builtin one. The pattern isn't *that* important but [improves compatibility](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#handling_browser_support) with browsers that don't support `type=date`.